opm/input/eclipse/Schedule/UDQ/UDQASTNode.cpp

```diff
@@ -669,6 +691,117 @@ void UDQASTNode::func_tokens(std::set<UDQTokenType>& tokens) const
     }
 }
 
+void UDQASTNode::populateRequiredObjects(UDQ::RequisiteEvaluationObjects& objects) const
+{
+    const auto vectorType = UDQ::targetType(std::get<std::string>(this->value));
+
+    if (this->selector.empty() && (vectorType != UDQVarType::REGION_VAR)) {
+        // No specific objects identified for this node.  Will generate an
+        // appropriately sized UDQ set and populate all elements of the set.
+        // Nothing to do.
+        //
+        // Note empty selector exception in the case of REGION_VARs.  Region
+        // level vectors encode the region set into the vector name, so we
+        // always have to run the full requisite object analysis for region
+        // level vectors.
+        return;
+    }
+
+    // If we get here, then the node identifies one or more specific objects
+    // for which to generate values in the resulting UDQ set.  Tell caller
+    // about these objects.
+
+    switch (vectorType) {
+    case UDQVarType::REGION_VAR:
+        this->populateRequiredRegionObjects(objects);
+        break;
+
+    case UDQVarType::SEGMENT_VAR:
+        this->populateRequiredSegmentObjects(objects);
+        break;
+
+    case UDQVarType::WELL_VAR:
+        this->populateRequiredWellObjects(objects);
+        break;
+
+    case UDQVarType::GROUP_VAR:
+        this->populateRequiredGroupObjects(objects);
+        break;
+
+    default:
+        // Nothing to do since either FIELD_VAR or unsupported.
+        break;
+    }
+}
+
+void UDQASTNode::populateRequiredGroupObjects(UDQ::RequisiteEvaluationObjects& objects) const
+{
+    // The selector is a group name like 'G' or a group name root like 'G*'.
+    assert (this->selector.size() == 1);
+
+    objects.groups.insert(this->selector.front());
+}
+
+namespace {
+    std::optional<int> parsePositiveInt(std::string_view s)
+    {
+        auto result = 0;
+        auto [ptr, ec] { std::from_chars(s.data(), s.data() + s.size(), result) };
+
+        if ((ec == std::errc{}) && (ptr == s.data() + s.size()) && (result > 0)) {
+            // s is "7" or similar.
+            return { result };
+        }
+
+        return {};
+    }
+} // Anonymous namespace
+
+void UDQASTNode::populateRequiredRegionObjects(UDQ::RequisiteEvaluationObjects& objects) const
+{
+    const auto split = std::string::size_type{5};
+
+    const auto& vector = std::get<std::string>(this->value);
+    if (vector.size() <= split) {
+        // No specific region set, meaning this vector applies to all region
+        // sets.  Cannot enumerate requisite objects.
```

Comment:
maybe 'no need to' or 'No requisite objects'

Reply:
> maybe 'no need to' or 'No requisite objects'

I switched to "no requisite objects".